esp32-mock-generator/src/main.cpp

```
@@ -0,0 +1,319 @@
+// ============================================================================
+// WiFi Connection
+// ============================================================================
+void connectWiFi() {
+  Serial.printf("→ Connecting to WiFi: %s ", WIFI_SSID);
+
+  WiFi.mode(WIFI_STA);
+  WiFi.begin(WIFI_SSID, WIFI_PASSWORD);
+
+  int attempts = 0;
+  while (WiFi.status() != WL_CONNECTED && attempts < 30) {
+    delay(500);
```

Clang (14.0.6)

[warning] 500 is a magic number; consider replacing it with a named constant

(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)